core/sr-io/src/lib.rs

```diff
@@ -205,15 +205,20 @@ export_api! {
 
 		/// Verify and recover a SECP256k1 ECDSA signature.
 		/// - `sig` is passed in RSV format. V should be either 0/1 or 27/28.
-		/// - returns `Err` if the signature is bad, otherwise the 64-byte pubkey (doesn't include the 0x04 prefix).
+		/// - returns `Err` if the signature is bad, otherwise the 64-byte raw pubkey (doesn't include the 0x04 prefix).
 		fn secp256k1_ecdsa_recover(sig: &[u8; 65], msg: &[u8; 32]) -> Result<[u8; 64], EcdsaVerifyError>;
+
+		/// Verify and recover a SECP256k1 ECDSA signature.
+		/// - `sig` is passed in RSV format. V should be either 0/1 or 27/28.
+		/// - returns `Err` if the signature is bad, otherwise the 33-byte compressed pubkey.
+		fn secp256k1_ecdsa_recover_compressed(sig: &[u8; 65], msg: &[u8; 32]) -> Result<[u8; 33], EcdsaVerifyError>;
```

Comment:
Why not merge both functions?

Reply:
how?

Reply:
Can we not just expose the `compressed` parameter?

Reply:
not without having a dynamic `enum` return type, which would hamper usability.

Reply:
not without having a dynamic `enum` return type, which would hamper usability.

Reply:
It's possible to leave these two API functions separate and still have a single Wasm host function that takes a `compressed` parameter (since that just returns a u32 status and writes bytes to the output).

Reply:
i just added the extra function - backwards compatible now.